zerver/tests/test_bots.py

```diff
@@ -326,7 +326,9 @@ def test_add_bot_email_address_visibility(self) -> None:
             RealmUserDefault.EMAIL_ADDRESS_VISIBILITY_ADMINS,
             acting_user=None,
         )
-        user.refresh_from_db()
+
+        # refresh user
+        user = self.example_user("hamlet")
```

Comment:
One theory is that we could have this be calling `self.example_user("hamlet", skip_cache=True)` or some similar abstraction for cases where we previously did `refresh_from_db`.

Reply:
No, again, we actually do want the memcached piece here.  We want to fetch the data in the same way that the server would.